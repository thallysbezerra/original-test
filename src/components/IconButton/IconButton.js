import React from 'react';
import PropTypes from 'prop-types';

import './IconButton.scss';

const IconButton = ({ className, color, disabled, icon, onClick, size }) => (
	<button
		className={`icon-button icon-button--${color} icon-button--${size} ${className}`}
		onClick={onClick}
		disabled={disabled && disabled}
	>
		<i className={`icon-button__icon fas fa-${icon}`} />
	</button>
);

IconButton.propTypes = {
	className: PropTypes.string,
	color: PropTypes.string,
	icon: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	size: PropTypes.string
};

IconButton.defaultProps = {
	className: '',
	color: 'primary',
	icon: 'question-circle',
	onClick: () => console.log('Do something'),
	size: 'medium'
};

export default IconButton;
