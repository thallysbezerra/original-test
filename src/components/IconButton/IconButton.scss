@import '../../css/Theme.scss';

.icon-button {
	align-items: center;
	border: 2px solid;
	border-radius: 50%;
	background-color: $white;
	display: flex;
	justify-content: center;
	outline: none;
	transition: background-color 0.15s ease-in-out,
		border-color 0.15s ease-in-out, color 0.15s ease-in-out;

	&:hover {
		cursor: pointer;
	}

	&:disabled {
		border-color: $grey-four;
		color: $grey-four;
		pointer-events: none;
	}

	&--small {
		@extend .icon-button;
		font-size: 18px;
		height: 35px;
		width: 35px;
	}

	&--medium {
		@extend .icon-button;
		font-size: 24px;
		height: 45px;
		width: 45px;
	}

	&--primary {
		border-color: $salmon;
		color: $salmon;

		&:hover {
			background-color: $salmon;
			border-color: $white;
			color: $white;
		}
	}

	&--secondary {
		border-color: $blue-dark;
		color: $blue-dark;

		&:hover {
			background-color: $blue-dark;
			border-color: $white;
			color: $white;
		}
	}
}
