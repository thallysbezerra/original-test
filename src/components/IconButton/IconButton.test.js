import React from 'react';
import IconButton from './IconButton';

import renderer from 'react-test-renderer';

describe('IconButton UI', () => {
	it('<IconButton /> must have the snapshot version.', () => {
		const tree = renderer.create(<IconButton />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
