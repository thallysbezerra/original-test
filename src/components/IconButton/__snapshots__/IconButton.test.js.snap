// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton UI <IconButton /> must have the snapshot version. 1`] = `
<button
  className="icon-button icon-button--primary icon-button--medium "
  onClick={[Function]}
>
  <i
    className="icon-button__icon fas fa-question-circle"
  />
</button>
`;
